### DISCUSSION TITLE
Bubble self-energy

### DISCUSSION THREAD

**Discussion by apandit2021** [ttadano/alamode#38] [Q&A]
Dear Dr. Tadano,

To calculate the bubble self-energy, I used the effective second-order harmonic force constants from the SCPH calculation and the third order force constants to evaluate the bubble-self energy (following the previous email discussion with Dr. Tadano). 
1) To make sure, could you please see my input file below and let me know if I need any corrections? 
2) Also, what is the difference between the bubble self-energy and bubble free-energy? What is the specific usefulness of bubble free-energy, and how can we calculate it using ALAMODE? 
 
&general
PREFIX = scph_bubble
MODE = RTA
FC2XML = harmonic-2-300K.xml
FCSXML =cubic_Opt.xml
NKD = 2
KD = * *
NONANALYTIC = 2
BORNINFO = *.born
EMIN = -10; EMAX = 400 ; DELTA_E = 0.1
TMIN = 0; TMAX = 700
DT = 50
/

&scph
SELF_OFFDIAG = 0
MAXITER = 8000
MIXALPHA = 0.05
KMESH_INTERPOLATE = 5 5 1
KMESH_SCPH = 10 10 2
TOL_SCPH = 1.0e-8
/

&cell

50.00 # factor in Bohr unit
..........

/

&kpoint
2
80 80 1
/

&analysis
KS_INPUT = mode_TA12.in
SELF_W = 1
#SPS = 1
/

and mode_TA12.in looks as
4
0.0 0.0 0.0 1
0.5 0.0 0.0 1
0.0 0.0 0.0 2
0.5 0.0 0.0 2

Best,
Abhiyan

**Comment by apandit2021** [comment]
Dear Dr. @ttadano ,

In case you forgot or did not see, could you please respond to my above queries at your convenience?
Thank you very much for your help.

Best,
Abhiyan

**Comment by AbhiyanPandit** [comment]
Dear Dr. Tadano,

I want to re-confirm about the calculation of the bubble self energy calculation, which we discussed via email previously. To get the renormalized frequency, I use equation: 
/omega^2 = \Omega_{scph}^2 - 2*\Omega_{scph}*Re\(Self_bubble(0))
So, in the PREFIX.Self.* output file (whose top part looks as below), please correct me if followings are not correct.
2nd column --> \Omega_{scph}^2
4th column --> Re\(Self_bubble), Re\(Self_bubble(0)) --> first row of the 4th column 
**And as I check, there is a same value of \Omega_{scph}^2 for a specific temperature. It means I can look just the first row of 2nd column for \Omega_{scph}^2 value. So then what is the usefulness/difference of setting E_min, E_max, and DELTA_E values for the calculation of \omega^2 in above equation?**

$ more bubble_energy.Self.1
# xk =               0              0              0
# mode = 1
## T[K], Freq (cm^-1), omega (cm^-1), Self.real (cm^-1), Self.imag (cm^-1)
         0    1.09737e-10         0              0              0
         0    1.09737e-10         1              0              0
         0    1.09737e-10         2              0              0
         0    1.09737e-10         3              0              0
         0    1.09737e-10         4              0              0
         0    1.09737e-10         5              0              0
         0    1.09737e-10         6              0              0
         0    1.09737e-10         7              0              0
         0    1.09737e-10         8              0              0
         0    1.09737e-10         9              0              0
         0    1.09737e-10        10              0              0
         0    1.09737e-10        11              0              0
         0    1.09737e-10        12              0              0
         0    1.09737e-10        13              0              0
         0    1.09737e-10        14              0              0
         0    1.09737e-10        15              0              0
         0    1.09737e-10        16              0              0
         0    1.09737e-10        17              0              0
         0    1.09737e-10        18              0              0
         0    1.09737e-10        19              0              0
         0    1.09737e-10        20              0              0
         0    1.09737e-10        21              0              0
         0    1.09737e-10        22              0              0
         0    1.09737e-10        23              0              0
         0    1.09737e-10        24              0              0
         0    1.09737e-10        25              0              0
         0    1.09737e-10        26              0              0
....................................
.....................................

Best,
Abhiyan

**Reply by ttadano** [reply]
About the meaning of each column/row, your understanding is correct. 

> So then what is the usefulness/difference of setting E_min, E_max, and DELTA_E values for the calculation of \omega^2 in above equation?

In this option (`SELF_W=1`), the frequency-dependent bubble self-energy is computed, which is useful for computing the phonon spectral function. Currently, you only need the omega=0 (static) value of the self-energy. So, the omega/=0 data looks unnecessary. However, in the current implementation, the real part of the self-energy is computed from the imaginary part by using the Kramers-Kronig relation. This requires the imaginary part of the self-energy for the whole frequency range, which must be correctly specified by `EMIN`, `EMAX`, and `DELTA_E`. Most importantly, `EMAX` should be 2x larger than the maximum phonon frequency of the system.

**Reply by djasrasaria** [reply]
Is there any documentation of the SELF_W and related keywords? I do not see this on any of the documentation. Thank you!
